### ISSUE TITLE
Z_HOMING_HEIGHT - Possible Improvement?

### ISSUE THREAD

**Issue by 1Alfa1** [MarlinFirmware/Marlin#17095]
###
Working with MPCNC and MARLIN 414D Bugfix 2.0.x for dual endstops flavor from V1Engineering.

After much work to get Z_HOMING_HEIGHT working through GITHUB and other research:
notes on my parameters:
changed Z_MAX_ENDSTOP_INVERTING to true (THANK YOU BLAIRQ!)
changed in G28.cpp
//destination.z = z_homing_height;
      destination.z = TEST(axis_known_position, Z_AXIS) ? z_homing_height : current_position.z + z_homing_height;
Didn't go back to original "destination.z = z_homing_height"  due to what I read about this.
*Side note - see no difference with the below, (What was this about?) 
destination.z = z_homing_height + (TEST(axis_known_position, Z_AXIS) ? 0.0f : current_position.z);

I am using Z_SAFE_HOMING and have my Z touch sensor on other side of bed, also using probe feature with a servo idea I developed to avoid using alligator clip on endmill, to touch top of spindle armature because case is no conductivity / not reliable. I have other changes as well but I don't think any are specific to the need/benefit for this improvement.

Change G28.cpp to include z in IF statement on whether to do Raise Z before homing.
### Feature Workflow

1. Home All (unknown or with Z<Z_HOMIING_HEIGHT) and Z raise before home works
2. Move X and/or Y around and then Z lower than Z_HOMING_HEIGHT
3. Use custom command "HOME Z AXIS" and it could crash and endmill (or nozzle).
4. Change line in G28.cpp from:
if (z_homing_height && (doX || doY)) { 
to
 if (z_homing_height && (doX || doY || doZ)) {
5. Now Z is raised even if doing a Z only home.

#### Additional Information
I apologize in advance if I am not following protocol for issues, I am new to ALL of this. Just trying to contribute.

**Comment by thinkyhead** [comment]
> I apologize in advance if I am not following protocol for issues

Were the instructions in the issue template unclear? Your post is very hard to follow. If you would include your configurations as explicitly requested then we won't have to harass you with a hundred questions.

**Comment by thinkyhead** [comment]
So, here's my first of a hundred questions while we wait for your configurations.

Are you using `Z_SAFE_HOMING` and is your problem due to the fact that XY needs to do a sideways move before the Z axis can home? Or do you have some kind of probe that won't work if it starts out too low to the bed?

The Z raise is set up to do a raise over the current position when the current Z position is "unknown" and to do a raise to the set height when the Z position is "known."

**Comment by 1Alfa1** [comment]
[Configuration.txt](https://github.com/MarlinFirmware/Marlin/files/4304079/Configuration.txt)
[Configuration_adv.txt](https://github.com/MarlinFirmware/Marlin/files/4304080/Configuration_adv.txt)
the video just shows how I am using the Probe option it does not show the "undesired behaviour" that I speak of in step 3 of feature workflow.

**Comment by 1Alfa1** [comment]
Video not working.... here is picture
![20200303_200920_HDR](https://user-images.githubusercontent.com/61944071/76173777-b8a4cf80-6178-11ea-9818-4e34fddb43b0.jpg)

**Comment by 1Alfa1** [comment]
Let me try to word this better. I know I added too much irrelevant info but I had so much trouble getting other things to work first, I thought it would be helpful. If I home and then move the head around manually and lower the head below the 0 position.. i.e below the touch sensor which is this (https://www.amazon.com/Z-Axis-Router-Touch-Setting-Milling/dp/B07GT3CHWZ/ref=sr_1_2?keywords=touch+plate+for+CNC+machine&qid=1583764743&sr=8-2). Then if I do a "Z HOME" (i.e. not re home x or y). It will crash. BUT if you add || doZ to the right place in the G28.cpp code... it appears to fix that problem. (Not sure if it would cause another issue).
![Touchplate_](https://user-images.githubusercontent.com/61944071/76225382-d0209e80-61f2-11ea-9f52-9f9b222e8659.jpg) Thank you to all the creators Marlin is awesome!

**Comment by boelle** [comment]
@1Alfa1 is this still an issue? or is this purely support questions?
@thinkyhead

**Comment by boelle** [comment]
This Issue Queue is for Marlin bug reports and development-related issues, and we prefer not to handle user-support questions here. (As noted on [this page](https://github.com/MarlinFirmware/Marlin/blob/2.0.x/.github/contributing.md#i-dont-want-to-read-this-whole-thing-i-just-have-a-question).) For best results getting help with configuration and troubleshooting, please use the following resources:

- MarlinFW.org [Marlin Documentation](http://marlinfw.org)
- RepRap.org [Marlin Forum](http://forums.reprap.org/list.php?415)
- [Tom's 3D Forums](https://discuss.toms3d.org/)
- Facebook Group ["Marlin Firmware"](https://www.facebook.com/groups/1049718498464482/)
- Facebook Group ["Marlin Firmware for 3D Printers"](https://www.facebook.com/groups/3Dtechtalk/)
- [Marlin Configuration](https://www.youtube.com/results?search_query=marlin+configuration) on YouTube
- Marlin Discord server. Join link: https://discord.gg/n5NJ59y

After seeking help from the community, if the consensus points to to a bug in Marlin, then you should post a [bug report](https://github.com/MarlinFirmware/Marlin/issues/new/choose).
